handler/src/main/java/io/netty/handler/ssl/Java9SslUtils.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.ssl;
+
+import static io.netty.util.internal.ObjectUtil.checkNotNull;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.SSLParameters;
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.security.PrivilegedExceptionAction;
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+import io.netty.util.internal.EmptyArrays;
+import io.netty.util.internal.PlatformDependent;
+import io.netty.util.internal.StringUtil;
+import io.netty.util.internal.logging.InternalLogger;
+import io.netty.util.internal.logging.InternalLoggerFactory;
+
+final class Java9SslUtils {
+    private static final InternalLogger log = InternalLoggerFactory.getInstance(Java9SslUtils.class);
+    private static final Method SET_APPLICATION_PROTOCOL;
+    private static final Method GET_APPLICATION_PROTOCOL;
+    private static final Method SET_HANDSHAKE_APPLICATION_PROTOCOL_SELECTOR;
+    private static final Class<?> BI_FUNCTION_CLASS;
+
+    static {
+        Method setter = null;
+        Method getter = null;
+        Method selector = null;
+        Class<?> biFunction = null;
+
+        int version = PlatformDependent.javaVersion();
+        if (version >= 9) {
+            try {
+                SSLEngine engine = SSLContext.getDefault().createSSLEngine();
```

Comment:
Hmm... that is even more troublesome. Yeah, swapping to `SSLContext.getInstance("TLS")` is enough at this point.